src/reach_study.cpp

```diff
@@ -260,6 +270,16 @@ void runReachStudy(const YAML::Node& config, const std::string& config_name, con
   if (opt_config["max_threads"])
     params.max_threads = opt_config["max_threads"].as<std::size_t>();
 
+  // read the initial joint positions if specified
+  const YAML::Node seed_state_yaml = opt_config["seed_state"];
+  for (auto it = seed_state_yaml.begin(); it != seed_state_yaml.end(); ++it)
+  {
+    const YAML::Node& seed_state_entry = *it;
+    std::string name = reach::get<std::string>(seed_state_entry, "name");
+    double position = reach::get<double>(seed_state_entry, "position");
+    params.seed_state.insert(std::pair<std::string, double>(name, position));
+  }
```

Comment:
We should probably also check that the names of the joints provided in `params.seed_state` contain at least all of the joint names that the IK solver interface knows about (could contain more in the case that the reach study motion group is just a subset of all the movable joints, e.g., robot on a gantry/mobile platform). I think we should also move the setting of the default seed state into this function as well for clarity, so everything about `params.seed_state` is set in one place

I would suggest moving this block below the initialization of the IK solver interface, and doing something like the following

```suggestion
  // Load the IK solver plugin
  ....

  // read the initial joint positions if specified
  const YAML::Node seed_state_config = opt_config["seed_state"];
  if(seed_state_config.isNull())
  {
    // Initialize with all zeros
    ...
  }
  else
  {
    for (auto it = seed_state_yaml.begin(); it != seed_state_yaml.end(); ++it)
    {
      const YAML::Node& seed_state_entry = *it;
      std::string name = reach::get<std::string>(seed_state_entry, "name");
      double position = reach::get<double>(seed_state_entry, "position");
      params.seed_state.insert(std::pair<std::string, double>(name, position));
    }
    
    // Check that a seed state is defined for all joints known to the IK interface
    const std::vector<std::string> joint_names = ik_solver->getJointNames();
    for(const std::string& name : joint_names)
      if(params.seed_state.find(name) == params.seed_state.end())
        throw std::runtime_error("Seed state parameter does not include joint '" + name + "'");
  }
```